script.recap/resources/lib/viewer.py

```diff
@@ -0,0 +1,165 @@
+            # Page down is going to the next page
+            self.onClick(Viewer.NEXT_BUTTON)
+
+    def onClick(self, controlID):
+        # Play button has been clicked
+        if controlID == Viewer.CLOSE_BUTTON:
+            log("Viewer: Close click action received: %d" % controlID)
+            self.close()
+        elif controlID == Viewer.NEXT_BUTTON:
+            log("Viewer: Request to view next image: %d" % controlID)
+            self._setSlideshowImage(self.currentImage + 1)
+        elif controlID == Viewer.PREVIOUS_BUTTON:
+            log("Viewer: Request to view previous image: %d" % controlID)
+            self._setSlideshowImage(self.currentImage - 1)
+
+    def close(self):
+        log("Viewer: Closing window")
+        self.isClosedFlag = True
+        xbmcgui.WindowXMLDialog.close(self)
+
+    def isClosed(self):
+        return self.isClosedFlag
+
+    def showNextSlideshowImage(self):
+        autoNextImage = self.currentImage + 1
+        # If we are already on the last image, check if we loop or not
+        if autoNextImage > (len(self.slideshow) - 1):
+            if Settings.isLoopSlideshow():
+                autoNextImage = 0
+            else:
+                return
+
+        # Move on to the next image in the slideshow
+        self._setSlideshowImage(autoNextImage)
+
+    def _setSlideshowImage(self, idx):
+        log("Viewer: Showing image %d" % idx)
+        if idx < 0:
+            if not Settings.isLoopSlideshow():
+                log("Viewer: Image %d out of range 0 to %d" % (idx, len(self.slideshow)))
+            idx = len(self.slideshow) - 1
+            log("Viewer: Image looped to last image %d" % idx)
+        elif idx > (len(self.slideshow) - 1):
+            if not Settings.isLoopSlideshow():
+                log("Viewer: Image %d out of range 0 to %d" % (idx, len(self.slideshow)))
+            idx = 0
+            log("Viewer: Image looped to first image %d" % idx)
+
+        self.currentImage = idx
+        imageControl = self.getControl(Viewer.IMAGE_ID)
+        imageControl.setImage(self.slideshow[idx]["image"])
+
+        if Settings.showImageCaptions():
+            captionControl = self.getControl(Viewer.CAPTION_ID)
+            captionControl.setText(self.slideshow[idx]["caption"])
+
+        previousControl = self.getControl(Viewer.PREVIOUS_BUTTON)
+        nextControl = self.getControl(Viewer.NEXT_BUTTON)
+        # For automated slideshows, do not show the next and previous buttons
+        if Settings.isAutoSlideshow():
+            previousControl.setVisible(False)
+            nextControl.setVisible(False)
+        # If we are looping then need to always show the next and previous buttons
+        elif Settings.isLoopSlideshow():
+            previousControl.setVisible(True)
+            nextControl.setVisible(True)
+        else:
+            if idx < 1:
+                previousControl.setVisible(False)
```

Comment:
I'm a bit old-school, also, there is a more complex comparison about 5 lines below it, and this way there is a certain amount of symmetry to the code